core/utxo/group_chain.go

```diff
@@ -0,0 +1,105 @@
+package utxo
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/xuperchain/xuperchain/core/contract"
+	"github.com/xuperchain/xuperchain/core/xmodel"
+)
+
+func (uv *UtxoVM) QueryChainInList() map[string]bool {
+	return uv.getChainInList()
+}
+
+func (uv *UtxoVM) QueryIPsInList(bcname string) map[string]bool {
+	return uv.getIPsInList(bcname)
+}
+
+func (uv *UtxoVM) getIPsInList(bcname string) map[string]bool {
+	ipMap := map[string]bool{}
+	args := map[string][]byte{
+		"bcname": []byte(bcname),
+	}
+
+	groupChainContract := uv.GetGroupChainContract()
+	if groupChainContract == nil {
+		return ipMap
+	}
+	moduleName := groupChainContract.ModuleName
+	contractName := groupChainContract.ContractName
+	methodName := groupChainContract.MethodName + "Node"
+
+	fmt.Println("============>", "moduleName:", moduleName, "contractName:", contractName, "methodName:", methodName)
+
+	if moduleName == "" && contractName == "" && methodName == "" {
+		return ipMap
+	}
+
+	status, target, err := uv.queryGroupChain(moduleName, contractName, methodName, args)
+	if status >= 400 || err != nil || string(target) == "" {
+		return ipMap
+	}
+	res := strings.Split(string(target), "\x01")
+	for _, v := range res {
+		ipMap[v] = true
+	}
+
+	return ipMap
+}
+
+func (uv *UtxoVM) getChainInList() map[string]bool {
```

Comment:
I think chain node group is a xchaincore feature more than UTXO.

Reply:
Application state is related to utxo, so i support an api based on utxo.